--- a/ecl-16.1.2/src/c/numbers/atan.d
+++ b/ecl-16.1.2/src/c/numbers/atan.d
@@ -24,68 +24,6 @@
 
 #pragma STDC FENV_ACCESS ON
 
-static double
-ecl_atan2_double(double y, double x)
-{
-        if (signbit(x)) {
-                if (signbit(y)) {
-                        return -ECL_PI_D + atan(-y / -x);
-                } else if (y == 0) {
-                        return ECL_PI_D;
-                } else {
-                        return ECL_PI_D - atan(y / -x);
-                }
-        } else if (x == 0) {
-                if (signbit(y)) {
-                        return -ECL_PI2_D;
-                } else if (y == 0) {
-                        return x / y;  /* Produces a NaN */
-                } else {
-                        return ECL_PI2_D;
-                }
-        } else {
-                if (signbit(y)) {
-                        return -atan(-y / x);
-                } else if (y == 0) {
-                        return (double)0;
-                } else {
-                        return atan(y / x);
-                }
-        }
-}
-
-#ifdef ECL_LONG_FLOAT
-static long double
-ecl_atan2_long_double(long double y, long double x)
-{
-        if (signbit(x)) {
-                if (signbit(y)) {
-                        return -ECL_PI_L + atanl(-y / -x);
-                } else if (y == 0) {
-                        return ECL_PI_L;
-                } else {
-                        return ECL_PI_L - atanl(y / -x);
-                }
-        } else if (x == 0) {
-                if (signbit(y)) {
-                        return -ECL_PI2_L;
-                } else if (y == 0) {
-                        return x / y;  /* Produces a NaN */
-                } else {
-                        return ECL_PI2_L;
-                }
-        } else {
-                if (signbit(y)) {
-                        return -atanl(-y / x);
-                } else if (y == 0) {
-                        return (long double)0;
-                } else {
-                        return atanl(y / x);
-                }
-        }
-}
-#endif
-
 cl_object
 ecl_atan2(cl_object y, cl_object x)
 {
@@ -98,13 +36,12 @@
         if (tx < ty)
                 tx = ty;
         if (tx == t_longfloat) {
-                long double d = ecl_atan2_long_double(ecl_to_long_double(y),
-                                                      ecl_to_long_double(x));
+      long double d = atan2l(ecl_to_long_double(y), ecl_to_long_double(x));
                 output = ecl_make_long_float(d);
         } else {
                 double dx = ecl_to_double(x);
                 double dy = ecl_to_double(y);
-                double dz = ecl_atan2_double(dy, dx);
+      double dz = atan2(dy, dx);
                 if (tx == t_doublefloat) {
                         output = ecl_make_double_float(dz);
                 } else {
@@ -114,7 +51,7 @@
 #else
         double dy = ecl_to_double(y);
         double dx = ecl_to_double(x);
-        double dz = ecl_atan2_double(dy, dx);
+    double dz = atan2(dy, dx);
         if (ECL_DOUBLE_FLOAT_P(x) || ECL_DOUBLE_FLOAT_P(y)) {
                 output = ecl_make_double_float(dz);
         } else {
